content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md

```diff
@@ -0,0 +1,242 @@
+* Using cURL
+* Using Python
+* Using Go
+* Using PowerShell
+* Using Ansible, Terraform, etc.
+
+The [HPE GreenLake for Compute Ops Management API Reference site](https://developer.greenlake.hpe.com/docs/greenlake/services/compute-ops/public/openapi/compute-ops-latest/overview/) leverages an OpenAPI conformant documentation that provides a complete explanation of the operations supported by the Unique Resource Identifiers [(URIs)](<>)as well as sample requests and responses.
+
+# Using PowerShell
+
+From a Terminal application, I start a PowerShell environment and set some variables that will be used later. 
+
+```powershell
+# Store our Compute Ops Manager API endpoint
+$endpoint = "https://us-west2-api.compute.cloud.hpe.com"
+# Initialize an empty headers array
+$headers = @{} 	
+# Add a header for authorization, using the Access Token generated in task 2	
+$headers["Authorization"] = "Bearer_access_token_here"
+```
+
+![figure3](/img/greenlake-com-ops-api-pyt-pow1.png)
+
+Our first request:
+
+```powershell
+$response = Invoke-webrequest "$endpoint/compute-ops/v1beta2/servers?limit=2" -Method GET -Headers $headers
+```
+
+When the request is successful, the response should display 200:
+
+![figure4](/img/greenlake-com-ops-api-pyt-pow2.png)
+
+```powershell
+$response.StatusCode
+```
+
+The JSON data returned by the request is in $response.Content but it is rather hard to read as is, I can prettify it a bit with: 
+
+```powershell
+$response.Content | ConvertFrom-Json | Format-List
+```
+
+Now, the request returned 2 objects (count : 2) and the actual server details are in the items array property.
+
+![figure5](/img/greenlake-com-ops-api-pyt-pow3.png)
+
+The API supports several query parameters depending on the resource type, such as filter, limit (maximum number of records to return), offset (resource offset to start the response from) and sort (order in which to return the resources in the collection).
+
+Filters provide the ability to limit the resources that take part in the action of a REST call. When a REST call includes a filter, the GET or DELETE action is restricted to a response that meets the filter requirements. Filters are specified by using the query parameter 'filter'.
+
+A simple example of filtering follows:
+
+**GET <URI>?filter=powerState eq 'Off'**
+
+This example shows a simple filter. The resources returned by the query are limited to results with the attribute powerState and the value Off. 
+
+To use a filter on a nested property name, the '/' separator can be specified as follows:
+
+**GET /compute-ops/v1beta2/servers?filter=hardware/model eq 'ProLiant DL365 Gen10 Plus'**
+
+To test this filter, I run  the following command: 
+
+```powershell
+$response = Invoke-webrequest "$endpoint/compute-ops/v1beta2/servers?filter=contains(hardware/model,'DL365')" -Method GET -Headers 
+$headers
+```
+
+I start by checking the request was successful as before:
+
+```powershell
+$response.StatusCode
+```
+
+it should display 200
```

Comment:
This line should read "It should display 200." (period at the end and initial capital letter at the beginning of the sentence)